lib/services/role.go

```diff
@@ -1887,69 +1600,174 @@ func (m *DatabaseNameMatcher) String() string {
 	return fmt.Sprintf("DatabaseNameMatcher(Name=%v)", m.Name)
 }
 
-// CheckAccessToDatabase checks if this role set has access to a particular database.
-//
-// The checker always checks the server namespace, other matchers are supplied
-// by the caller.
-func (set RoleSet) CheckAccessToDatabase(server types.Database, mfa AccessMFAParams, matchers ...RoleMatcher) error {
+type loginMatcher struct {
+	login string
+}
+
+// NewLoginMatcher creates a RoleMatcher that checks whether the role's logins
+// match the specified condition.
+func NewLoginMatcher(login string) RoleMatcher {
+	return &loginMatcher{login: login}
+}
+
+func (l *loginMatcher) Match(role types.Role, typ types.RoleConditionType) (bool, error) {
+	logins := role.GetLogins(typ)
+	for _, login := range logins {
+		if l.login == login {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// AccessCheckable is the subset of types.Resource required for the RBAC checks.
+type AccessCheckable interface {
+	GetKind() string
+	GetName() string
+	GetMetadata() types.Metadata
+	GetAllLabels() map[string]string
+}
+
+func rbacDebugLogger() (debugEnabled bool, debugf func(format string, args ...interface{})) {
```

Comment:
Nit: Explain with a godoc the purpose of this method.

Reply:
There's an explanation at the call site, which is where other reviewers requested it, but I'm happy to move it.

Reply:
Yeah I saw it, I think it's worth explaining it here as well (e.g. so somebody wanting to know what this does doesn't have to search for where it's called).